Development.md

```diff
@@ -0,0 +1,250 @@
+
+This guide documents how to:
+
+* Set up a sandbox development Rack to develop and test infrastructure template changes
+* Set up a local development VM to run, develop and test API changes
+* Run the Rack unit test suite locally
+* Submitting changes upstream
+* Run an AWS integration test suite locally or on a CI server
+* Releasing artifacts to enable `convox rack update`
+
+## Sponsorship and Mentoring
+
+Developing Rack will incur AWS costs. If this is an obstacle for you to contribute you can contact support@convox.com to request sponsorship.
+
+Much of the technical design and implementation in Rack requires understanding of AWS, Docker, Golang, systems engineering and more. If you would like to better learn these systems to contribute, you can contact support@convox.com, join the [Public Slack](http://invite.convox.com/), or open up issues on [GitHub](http://github.com/convox/rack) to ask questions and/or request a mentor.
+
+## Development Rack Install
+
+Rack consumes numerous AWS and Docker APIs. The easiest way to develop Rack is with real AWS access keys interacting with real AWS resources like a Dynamo Table, ECS Cluster, and CloudFormation Stack.
+
+This is easy to bootstrap with the Rack project itself:
+
+```
+$ convox install --stack-name dev
+```
+
+It can also be bootstrapped with no `convox` tools via the AWS CLI:
+
+```
+$ aws cloudformation create-stack --stack-name dev --template-body file://$(pwd)/api/dist/kernel.json
+```
+
+You can also use any existing Rack with the caveat that running a local Rack against it could have side effects like terminating instances.
+
+## Development Rack Ingress
+
+If your changes also interact with Docker, you will want to open up access to the instance Docker daemons from your laptop.
+
+* Open the [Security Group Management Console](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#SecurityGroups)
+* Select the Security Group with the Group Name like "dev-SecurityGroup-4PNOYR5HUH83" and the Description "Instances"
+* Click the Inbound tab, then the Edit button, then the Add Rule button
+* Keep Custom TCP Rule and TCP protocol
+* Add "2376" for Port Range, and select "My IP" for Source
+* Click the Save button
+
+Warning: Do not expose instance port 2376 to "Anywhere"! This will expose your Docker daemons to the whole Internet.
+
+## Rack Golang Project
+
+Rack is written in Golang. To setup a Go environment, see the excellent [Getting Started](https://golang.org/doc/install) docs. You can then clone and build the project with the `go get` tool:
+
+```
+$ go get github.com/convox/rack/...
+```
+
+After this, `which convox` should refer to `$GOPATH/bin/convox`.
+
+## Development Rack Environment
+
+The development Rack is running an API process that has AWS Access Keys, AWS resource names, and other various settings in its environment. You need to copy this to your laptop
```

Comment:
This is slightly confusing. 
As i will be "develop"ing locally. I think the term "development rack" is slightly overloaded.
How about Laptop/Local rack and AWS/Cloud rack